### DISCUSSION TITLE
First time custom-fact question

### DISCUSSION THREAD

**Discussion by WembleyFord2** [puppetlabs/community#71] [Troubleshooting] [answered]
Total ruby noob and first time trying to right a custom fact.  It does look too complication and I've sort of got this working.. But one question - from the puppet documentation gives two ways of using a 'confine'. 

This lets me custom fact produce output  via facter on my gentoo box as expected
``
confine :kernel => "Linux"
``

But this produces no output -  
``
confine :kernel do |value|
    value == "Linux"
end
``
What am I failing to understand- the documentation would suggest both should work? 
Thanks in advance. 
Dave

**Comment by mhashizume** [comment, marked as answer]
Hi, this was a bug that was addressed in https://github.com/puppetlabs/facter/pull/2705. The fix will be included in the next release of Facter.
